### PR TITLE
Refactor(SPV-1217) Remove_TransportWrapper_and_Verify_OnAfterResponse

### PR DESCRIPTION
**Pull request by ac4ch**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: internal/restyutil/restyutil_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/restyutil"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+ What is the test doing?
```
Comment (dorzepowski) [inline internal/restyutil/restyutil_test.go]: This is pointless to add such a comments on tests.

File: internal/restyutil/restyutil_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+				"GET",
+				"http://mock-api/test",
+				httpmock.NewJsonResponderOrPanic(tc.statusCode, tc.responseBody),
+			)
+
+			// Make request
+			resp, err := client.R().Get("/test")
+
+			// Assert errors
+			if tc.expectedError != nil {
```
Comment (dorzepowski) [inline internal/restyutil/restyutil_test.go]: This adds a complexity to the test and also can lead to introduction of a bug in a test itself.

IMHO should extract single test for success response from the test cases list to a separated test.

File: internal/restyutil/restyutil_test.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+				require.Error(t, err)
+
+				var spvErr models.SPVError
+				if errors.As(err, &spvErr) && tc.expectedSPVError != nil {
+					require.Equal(t, *tc.expectedSPVError, spvErr)
+				} else {
+					require.Contains(t, err.Error(), tc.expectedError.Error())
+				}
```
Comment (dorzepowski) [inline internal/restyutil/restyutil_test.go]: We should rather use `require.ErrorIs` probably instead of this complex assertion.

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: internal/restyutil/restyutil_test.go (hunk 1 above)
Comment (ac4ch) [reply internal/restyutil/restyutil_test.go]: yes i was just explaining here what this test do .. i knew that we should remove it, before merge it

**Review by dorzepowski — APPROVED** [review]

**Review by mgosek-4chain — APPROVED** [review]

**Pull request merged**